scripts/smart-dump.ts

```diff
@@ -40,31 +42,30 @@ program.command('dump [recipe] [env]').action(async (recipe, env) => {
   const date = new Date().toISOString().substring(0, 10);
   const hash = md5(JSON.stringify({ entries, defaultDependencies, date })).slice(0, 5);
   const filename = `${date}.${hash}`;
-  let docs = [];
+  const seenModelRecords: Set<string> = new Set();
 
   let start = new Date();
-  logger.info('>>> Dumping...');
+  logger.info(`>>> Dumping... to dbdumps/${filename}.jsonl`);
+  const dumpFile = fs.createWriteStream(`dbdumps/${filename}.jsonl`);
   for (const entry of entries) {
     logger.info(`>>> Traversing DB for entry ${entries.indexOf(entry) + 1}/${entries.length}...`);
-    const newdocs = await traverse({ ...entry, defaultDependencies, parsed });
-    docs.push(...newdocs);
+    await traverse({ ...entry, defaultDependencies, parsed }, ei => {
+      const modelRecordKey = `${ei.model}.${ei.id}`;
+      if (!seenModelRecords.has(modelRecordKey)) {
+        dumpFile.write(JSON.stringify(ei) + os.EOL);
```

Comment:
Perhaps we should await the write to be done here.

Reply:
The writes are syncronous